### PR TITLE
{vis}[mixed] OpenCV 3.1.0 fix ippicv source download and library install (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

File: easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb [hunk 1]
```diff
@@ -34,6 +42,9 @@ dependencies = [
     ('GST-plugins-base', '1.6.4'),
 ]
 
+#The destination directory is "linux-<MD5>" see 3rdparty/ippicv/downloader.cmake
+preconfigopts = 'mkdir -p {0} && cp -a ../*.tgz {0}. &&'.format("3rdparty/ippicv/downloads/linux-808b791a6eac9ed78d32a7666804320e/")
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb]: @hajgato line is too long, how about this:

```python
ippicv_dir = '3rdparty/ippicv/downloads/linux-808b791a6eac9ed78d32a7666804320e/'
preconfigopts = "mkdir -p {0} && cp -a ../*.tgz {0} && ".format(ippicv_dir)
```

I'm not a big fan of using `.format` in easyconfigs, but admittedly this is not better:

```python
preconfigopts = "mkdir -p %s && cp -a ../*.tgz %s && " % (ippicv_dir, ippicv_dir)
```

That does follow the easyconfigs style we usually go with though...

File: easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb [hunk 2]
```diff
@@ -11,8 +11,16 @@ description = """OpenCV (Open Source Computer Vision Library) is an open source
...
-sources = ['%(version)s.zip']
+source_urls = [
+    'https://github.com/Itseez/opencv/archive/',
+    #The Hash is version dependent! see 3rdparty/ippicv/downloader.cmake
+    'https://raw.githubusercontent.com/Itseez/opencv_3rdparty/81a676001ca8075ada498583e4166079e5744668/ippicv',
+]
+
+sources = [
+    '%(version)s.zip',
+    ('ippicv_linux_20151201.tgz', 'cp %s %(builddir)s'),
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb]: @hajgato does this actually work? Don't you need to use `cp %s %%(builddir)s` or `cp %%s %(builddir)s` (not sure which template is resolved first)?

File: easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb [hunk 3]
```diff
@@ -52,6 +63,8 @@ configopts += '-DWITH_IPP=ON '
 configopts += '-DENABLE_SSE=ON -DENABLE_SSE2=ON -DENABLE_SSE3=ON '
 configopts += '-DWITH_CUDA=OFF '
 
+postinstallcmds = ["cp 3rdparty/ippicv/unpack/ippicv_lnx/lib/intel64/libippicv.*  %(installdir)s/lib"]
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb]: this should be reflected in `sanity_check_paths`? (also, remove the double space ;-))

File: easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb (hunk 2 above)
Comment (hajgato) [inline easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb]: @boegel Actually it worked for me. I have checked in the builddir, and the file was there.

File: easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb (hunk 1 above)
Comment (hajgato) [inline easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb]: @ boegel: OK

File: easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb (hunk 3 above)
Comment (hajgato) [inline easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb]: @boegel done

File: easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb]: hmm, that's pretty strange actually...

```
>>> "%s %(builddir)s" % {'builddir': 'test'}
"{'builddir': 'test'} test"
```

Comment (boegel) [inline easybuild/easyconfigs/o/OpenCV/OpenCV-3.1.0-foss-2016a.eb]: ah, no, it makes sense, the `resolve_template` method is smart enough to translate `%s` into `%%s` by itself before resolving the named templates, so fine as is

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2111.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/2cb523d213aa688103c9e913d1eb80b5 for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2625.swalot.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/d744f7d6927facad228dc770f35a9726 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @hajgato!

**Pull request merged**